src/emitMixin.ts

```diff
@@ -1,34 +1,53 @@
-import { getCurrentInstance } from 'vue'
+import { setDevtoolsHook, ComponentPublicInstance } from 'vue'
 
-export const attachEmitListener = () => {
-  return {
-    beforeCreate() {
-      let events: Record<string, unknown[]> = {}
-      ;(this as any).__emitted = events
-      const originalEmit = getCurrentInstance()!.emit
-      getCurrentInstance()!.emit = (event: string, ...args: unknown[]) => {
-        events[event]
-          ? (events[event] = [...events[event], [...args]])
-          : (events[event] = [[...args]])
+const enum DevtoolsHooks {
+  COMPONENT_EMIT = 'component:emit'
+}
+
+export const attachEmitListener = (vm: ComponentPublicInstance) => {
+  let events: Record<string, unknown[]> = {}
+  ;(vm as any).__emitted = events
+  // use devtools capture this "emit"
+  setDevtoolsHook(createDevTools(events))
+}
+
+function createDevTools(events) {
+  const devTools: any = {
```

Comment:
Instead of `any`, it would be great to properly type this. Ain't this a `DevtoolsHook` type?
Maybe there are not exposed by `vue-next` though?

Reply:
We can get it hackily (the interface itself is not available). not sure how beneficial it'll be:

```ts
import { devtools } from 'vue'

const blah: Partial<typeof devtools> = {
  emit: () => {}
}
```

May need to be a partial unless we want to define `on` and the other hooks (we could make them no-ops).